alerting/src/main/kotlin/com/amazon/opendistroforelasticsearch/alerting/transport/TransportAcknowledgeAlertAction.kt

```diff
@@ -0,0 +1,134 @@
+import org.elasticsearch.action.bulk.BulkResponse
+import org.elasticsearch.action.search.SearchRequest
+import org.elasticsearch.action.search.SearchResponse
+import org.elasticsearch.action.support.ActionFilters
+import org.elasticsearch.action.support.HandledTransportAction
+import org.elasticsearch.action.update.UpdateRequest
+import org.elasticsearch.client.Client
+import org.elasticsearch.common.inject.Inject
+import org.elasticsearch.common.xcontent.LoggingDeprecationHandler
+import org.elasticsearch.common.xcontent.NamedXContentRegistry
+import org.elasticsearch.common.xcontent.XContentFactory
+import org.elasticsearch.common.xcontent.XContentHelper
+import org.elasticsearch.common.xcontent.XContentParser
+import org.elasticsearch.common.xcontent.XContentParserUtils
+import org.elasticsearch.common.xcontent.XContentType
+import org.elasticsearch.index.query.QueryBuilders
+import org.elasticsearch.search.builder.SearchSourceBuilder
+import org.elasticsearch.tasks.Task
+import org.elasticsearch.transport.TransportService
+import java.time.Instant
+
+private val log = LogManager.getLogger(TransportAcknowledgeAlertAction::class.java)
+
+class TransportAcknowledgeAlertAction @Inject constructor(
+    transportService: TransportService,
+    val client: Client,
+    actionFilters: ActionFilters,
+    val xContentRegistry: NamedXContentRegistry
+) : HandledTransportAction<AcknowledgeAlertRequest, AcknowledgeAlertResponse>(
+        AcknowledgeAlertAction.NAME, transportService, actionFilters, ::AcknowledgeAlertRequest
+) {
+
+    override fun doExecute(task: Task, request: AcknowledgeAlertRequest, actionListener: ActionListener<AcknowledgeAlertResponse>) {
+        AcknowledgeHandler(client, actionListener, request).start()
+    }
+
+    inner class AcknowledgeHandler(
+        private val client: Client,
+        private val actionListener: ActionListener<AcknowledgeAlertResponse>,
+        private val request: AcknowledgeAlertRequest
+    ) {
+        val alerts = mutableMapOf<String, Alert>()
+
+        fun start() = findActiveAlerts()
+
+        private fun findActiveAlerts() {
+            val queryBuilder = QueryBuilders.boolQuery()
+                    .filter(QueryBuilders.termQuery(Alert.MONITOR_ID_FIELD, request.monitorId))
+                    .filter(QueryBuilders.termsQuery("_id", request.alertIds))
+            val searchRequest = SearchRequest()
+                    .indices(AlertIndices.ALERT_INDEX)
+                    .routing(request.monitorId)
+                    .source(SearchSourceBuilder().query(queryBuilder).version(true).seqNoAndPrimaryTerm(true))
+
+            client.search(searchRequest, object : ActionListener<SearchResponse> {
+                override fun onResponse(response: SearchResponse) {
+                    onSearchResponse(response)
+                }
+
+                override fun onFailure(t: Exception) {
+                    actionListener.onFailure(t)
+                }
+            })
```

Comment:
Just curious, why not continue using the `wrap` function for `ActionListener`, so it will be like:
`client.search(searchRequest, ActionListener<SearchResponse>.wrap(::onSearchResponse, ::onFailure))`

Reply:
I started this pattern mainly to customize exception and returned rest status with onFailure. Something like:
actionListener.onFailure(ElasticsearchStatusException("my error msg",RestStatus.INTERNAL_SERVER_ERROR))